File: server/FunctionalTests/Tests/Rojo04/RemoteCheckout.py
"""
TheNexusAvenger

Tests that the remote checkout request acts correctly.
"""

import NexusGitFunctionalTest



"""
Tests the remote checkout as a standalone request.
"""
class Rojo04TestRemoteCheckoutAsStandaloneRequest(NexusGitFunctionalTest.NexusGitFunctionalTest):
    """
    Setup for the test.
    """
    def setupTest(self):
        self.workspace.writeFile("rojo.json","{\"name\": \"Nexus Git Repository\",\"servePort\": 20013,\"partitions\": {\"src\": {\"path\": \"src\",\"target\": \"ReplicatedStorage.NexusGit\"},\"test\": {\"path\": \"test\",\"target\":\"ReplicatedStorage.NexusGitTest\"}}}")

        # Initialize the git repository.
        self.workspace.runCommand("git init")
        self.workspace.runCommand("git remote add origin https://github.com/TheBotAvenger/Initialized-Dummy-Repository.git")
        self.workspace.runCommand("git checkout -b master")
        self.workspace.runCommand("git pull origin master")

    """
    Runs the test.
    """
    def runTest(self):
        # Wait for it to initialize.
        self.setPortNumber(20013)
        self.waitForInitialization()

        # Checkout the gh-pages branch and assert the files are correct.
        self.sendPOSTRequest("/remotecheckout","{\"localBranch\":\"test_branch\",\"remote\":\"origin\",\"branch\":\"dummy-branch-2\"}")
        self.assertTrue(self.workspace.fileExists("DummyFile6"))
        self.assertFalse(self.workspace.fileExists("DummyFile3"))
        response = self.sendGETRequest("/listlocalbranches")
        self.assertEqual(response,"[\"master\",\"* test_branch\"]")

File: server/NexusGitTests/Functional/Rojo04/RemoteCheckout.cs
/*
 * TheNexusAvenger
 *
 * Tests the RemoteCheckout request with Rojo 0.4.
 */

using NUnit.Framework;

namespace NexusGitTests.Functional.Rojo04
{
    /*
     * Tests remote checkout as a standalone request.
     */
    public class TestRemoteCheckout : BaseFunctionalTest
    {
        /*
         * Sets up the test.
         */
        public override void Setup()
        {
            // Write the files.
            this.Workspace.WriteFile("rojo.json","{\"name\": \"Nexus Git Repository\",\"servePort\": 20001,\"partitions\": {\"src\": {\"path\": \"src\",\"target\": \"ReplicatedStorage.NexusGit\"},\"test\": {\"path\": \"test\",\"target\":\"ReplicatedStorage.NexusGitTest\"}}}");
            
            // Initialize the repository.
            this.Workspace.RunCommand("git","init");
            this.Workspace.RunCommand("git","remote add origin https://github.com/TheBotAvenger/Initialized-Dummy-Repository.git");
            this.Workspace.RunCommand("git","checkout -b master");
            this.Workspace.RunCommand("git","pull origin master");
        }
        
        /*
         * Runs the test.
         */
        public override void Run()
        {
            this.SendPOSTRequest("/remotecheckout","{\"localBranch\":\"test_branch\",\"remote\":\"origin\",\"branch\":\"dummy-branch-2\"}");
            Assert.IsTrue(this.Workspace.FileExists("DummyFile6"));
            Assert.IsFalse(this.Workspace.FileExists("DummyFile3"));
            var response = this.SendGETRequest("/listlocalbranches");
            Assert.AreEqual(response,"[\"master\",\"* test_branch\"]");
        }
    }
}